src/index/updater/inscription_updater.rs

```diff
@@ -76,66 +92,118 @@ impl<'a, 'db, 'tx> InscriptionUpdater<'a, 'db, 'tx> {
+      // skip subsidy since no inscriptions possible
       if tx_in.previous_output.is_null() {
         input_value += Height(self.height).subsidy();
-      } else {
-        for (old_satpoint, inscription_id) in
-          Index::inscriptions_on_output(self.satpoint_to_id, tx_in.previous_output)?
-        {
-          inscriptions.push(Flotsam {
-            offset: input_value + old_satpoint.offset,
-            inscription_id,
-            origin: Origin::Old { old_satpoint },
-          });
-        }
+        continue;
+      }
 
-        input_value += if let Some(value) = self.value_cache.remove(&tx_in.previous_output) {
-          value
-        } else if let Some(value) = self
-          .outpoint_to_value
-          .remove(&tx_in.previous_output.store())?
-        {
-          value.value()
-        } else {
-          self.value_receiver.blocking_recv().ok_or_else(|| {
-            anyhow!(
-              "failed to get transaction for {}",
-              tx_in.previous_output.txid
-            )
-          })?
-        }
+      // find existing inscriptions on input aka transfers of inscriptions
+      for (old_satpoint, inscription_id) in
+        Index::inscriptions_on_output(self.satpoint_to_id, tx_in.previous_output)?
+      {
+        let offset = input_value + old_satpoint.offset;
+        floating_inscriptions.push(Flotsam {
+          offset,
+          inscription_id,
+          origin: Origin::Old { old_satpoint },
+        });
+
+        inscribed_offsets.insert(offset);
       }
-    }
 
-    if inscriptions.iter().all(|flotsam| flotsam.offset != 0)
-      && Inscription::from_transaction(tx).is_some()
-    {
-      let flotsam = Flotsam {
-        inscription_id: txid.into(),
-        offset: 0,
-        origin: Origin::New {
-          fee: input_value - tx.output.iter().map(|txout| txout.value).sum::<u64>(),
-        },
+      let offset = input_value;
+
+      // different ways to get the utxo set (input amount)
+      input_value += if let Some(value) = self.value_cache.remove(&tx_in.previous_output) {
+        value
+      } else if let Some(value) = self
+        .outpoint_to_value
+        .remove(&tx_in.previous_output.store())?
+      {
+        value.value()
+      } else {
+        self.value_receiver.blocking_recv().ok_or_else(|| {
+          anyhow!(
+            "failed to get transaction for {}",
+            tx_in.previous_output.txid
+          )
+        })?
       };
 
-      if input_value == 0 {
-        self.update_inscription_location(
-          input_sat_ranges,
-          flotsam,
-          SatPoint {
-            outpoint: unbound_outpoint(),
-            offset: self.unbound_inscriptions,
+      // go through all inscriptions in this input
+      while let Some(inscription) = new_inscriptions.peek() {
+        if inscription.tx_in_index != u32::try_from(input_index).unwrap() {
+          break;
+        }
+
+        // ignore reinscriptions on already inscribed offset (sats)
```

Comment:
```suggestion
        // In this first part of the cursed inscriptions implementation we ignore reinscriptions. This will change once we implement reinscriptions.
```